combat-trainer.lic

```diff
@@ -4054,18 +4069,118 @@ class GameState
     @equipment_manager.stow_weapon(weapon_training[skill])
   end
 
-  def wield_offhand_weapon(skill)
+  # Wields a weapon to train the specified skill,
+  # but instead of inferring the weapon name from
+  # `weapon_training:` config, wields the specified
+  # weapon name given.
+  #
+  # Initially designed to support `ambush_stun` attacks
+  # where you need a specific weapon of a particular weight, etc.
+  # so the weapon you want to use may differ from your regular weapon
+  # for the same skill.
+  #
+  # @param weapon_name
+  #   The <adj> + <noun> of the weapon to wield as configured in your gear.
+  # @param skill
+  #   The weapon skill to train.
+  # @param hand
+  #   Which hand to equip the weapon into?
+  #   To equip offhand, specify 'left' or 'offhand'.
+  # @returns name of the wielded weapon
+  def wield_specific_weapon(weapon_name, skill, hand = 'main')
+    echo "wield_specific_weapon: weapon_name=#{weapon_name}, skill=#{skill}, hand=#{hand}" if $debug_mode_ct
+    if is_damaris_weapon?(weapon_name)
+      return wield_damaris_weapon(weapon_name, hand)
+    else
+      if hand =~ /^(left|off)/i
+        @equipment_manager.wield_weapon_offhand?(weapon_name, skill)
+      else
+        @equipment_manager.wield_weapon?(weapon_name, skill)
+      end
+    end
+    return weapon_name
+  end
+
+  # Wields a weapon to train the specified skill.
+  #
+  # @param skill
+  #   The weapon skill to train. We infer the weapon to wield based on this.
+  # @param hand
+  #   Which hand to equip the weapon into?
+  #   To equip offhand, specify 'left' or 'offhand'.
+  # @returns name of the wielded weapon
+  def wield_weapon(skill, hand = 'main')
     return if skill.eql?('Brawling')
     return if skill.eql?('Tactics')
 
-    @equipment_manager.wield_weapon_offhand(weapon_training[skill], skill)
-    weapon_training[skill]
+    hand = 'offhand' if skill =~ /Offhand Weapon/i
+    weapon_name = weapon_training[skill]
```

Comment:
Is this line the only difference between `wield_specific_weapon` and `wield_weapon`? If so, I'd prefer not to have the same code twice. I bet there's a way we can refactor this.